### PR TITLE
User registration

### PR DESCRIPTION
**Pull request by RostyslavStepanchuk**

### CODE REVIEW

**Review by skosinskiy — APPROVED** [review]

File: client/src/actions/auth.js [hunk 1]
```diff
@@ -39,17 +39,16 @@ export const loadUser = () => async dispatch => {
 
 // Register User
 
-export const register = ({ username, password }) => async dispatch => {
+export const register = (registerData) => async dispatch => {
```
Comment (tarasbashuk) [inline client/src/actions/auth.js:42]: Of course, it's cleaner, but I bet you will forget what kind of data we expecting in 1 week.

File: client/src/actions/auth.js [hunk 2]
```diff
@@ -39,17 +39,16 @@ export const loadUser = () => async dispatch => {
...
     headers: {
       'Content-Type': 'application/json'
     }
   }
 
-  const body = { username, password }
-
   try {
-    const res = await axios.post('/api/v1/users/sign-up', body, config)
+    console.log('sending request')
```
Comment (tarasbashuk) [inline client/src/actions/auth.js:50]: I suggest deleting consol.log whenever possible.

File: client/src/components/auth/ConfirmEmail.jsx [hunk 3]
```diff
@@ -1,19 +1,54 @@
...
+import {Link} from 'react-router-dom'
+
+import useStyles from './ConfirmEmailStyles'
 
 const ConfirmEmail = props => {
-    console.log(props)
+
+    const classes = useStyles();
+
+    const [ emailConfirmed, setEmailConfirmed ] = useState(false);
```
Comment (tarasbashuk) [inline client/src/components/auth/ConfirmEmail.jsx:13]: IMO it should be in store, not in the local state.

File: client/src/components/auth/ConfirmEmail.jsx [hunk 4]
```diff
@@ -1,19 +1,54 @@
...
+    const [ isError, setError] = useState(false);
+    const token = props.match.params.token;
+
+    axios.get('/api/v1/users/email/confirm/' + token)
+        .then(res => {
+            if(res.status === 200){
+                setEmailConfirmed(true)
+            }})
+        .catch(() => setError(true));
+
```
Comment (tarasbashuk) [inline client/src/components/auth/ConfirmEmail.jsx:23]: IMO it should be a separate action, which set confirm and email in the store.

File: client/src/components/auth/ConfirmEmail.jsx [hunk 5]
```diff
@@ -1,19 +1,54 @@
...
+    let content;
+    if (!isError) {
+        content = emailConfirmed ? 'Your email was confirmed. Welcome to the club!' : 'Here will be a spinner';
```
Comment (tarasbashuk) [inline client/src/components/auth/ConfirmEmail.jsx:26]: Don't get it about spinner.

File: client/src/actions/auth.js (hunk 2 above)
Comment (RostyslavStepanchuk) [reply client/src/actions/auth.js:50]: My bad. Just have forgotten to remove

**Pull request merged**